DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/EponymousPartGraphSyncer.cs

```diff
@@ -114,6 +120,68 @@ public class EponymousPartGraphSyncer : IContentPartGraphSyncer
             return Enumerable.Empty<ICommand>();
         }
 
+        public async Task<bool> VerifySyncComponent(ContentItem contentItem, INode node, ContentTypePartDefinition contentTypePartDefinition,
+            IEnumerable<IRelationship> relationships, IEnumerable<INode> destNodes)
+        {
+            foreach (var field in contentTypePartDefinition.PartDefinition.Fields)
+            {
+                JObject value = contentItem.Content[contentItem.ContentType][field.Name];
+
+                if (field.FieldDefinition.Name == "ContentPickerField")
+                {
+                    var relationshipType = $"ncs__has{field.Settings["ContentPickerFieldSettings"]!["DisplayedContentTypes"]![0]}";
+                    var contentItemIds = (JArray)value["ContentItemIds"]!;
+
+                    var contentCount = contentItemIds.Count;
+                    //TODO : need to ignore case for hasSocCode vs hasSOCCode - need to make sure these line up!
+                    var relationshipCount = relationships.Count(x => string.Equals(x.Type, relationshipType, StringComparison.CurrentCultureIgnoreCase));
+
+                    if (contentCount != relationshipCount)
+                    {
+                        return false;
+                    }
+
+                    foreach (var item in contentItemIds)
+                    {
+                        var contentItemId = (string)item!;
+
+                        var destContentItem = await _session
+                            .Query<ContentItem, ContentItemIndex>(x =>
+                                x.ContentItemId == contentItemId).FirstOrDefaultAsync();
+
+                        var destUri = (string)destContentItem.Content.GraphSyncPart.Text;
+
+                        var destNode = destNodes.SingleOrDefault(n => (string)n.Properties["uri"] == destUri);
+
+                        if (destNode == null)
+                        {
+                            return false;
+                        }
+
+                        var relationship = relationships.SingleOrDefault(x =>
+                            string.Equals(x.Type, relationshipType, StringComparison.CurrentCultureIgnoreCase) && x.EndNodeId == destNode.Id);
+
+                        if (relationship == null)
+                        {
+                            return false;
+                        }
+                    }
+                }
+                else
+                {
+                    var contentItemValue = value["Text"] ?? value["Html"] ?? value["Value"];
+                    var nodePropertyValue = node.Properties[$"ncs__{field.Name}"];
+
+                    if (Convert.ToString(contentItemValue) != Convert.ToString(nodePropertyValue))
```

Comment:
It'd probably be safer to handle numeric fields explicitly, rather than relying on a string comparison.
From memory, numeric fields with scale 0 are represented as floats in the content field, so show as e.g. "1.0", but we store them as integers in the graph, so would have the string "1".
It might actually work as is, but I think it would still be worth considering handling numerics explicitly.